File: src/main/python/extract_method.py
#Before

class PriceCalculator(object):
    def __init__(self, unit_price, num):
        self.unit_price = unit_price
        self.num = num

    @property
    def price(self):
        """Returns: total price, total price is
        tanka * num * tax(1.1)
        """
        return int(self.unit_price * self.num * 1.1)

#After

class PriceCalculator2(object):
    def __init__(self, unit_price, num):
        self.unit_price = unit_price
        self.num = num

    @property
    def price(self):
        """Returns: total price, total price is
        tanka * num * tax(1.1)
        """
        return self.add_tax(self.unit_price * self.num)

    def add_tax(self, total):
        """extract"""
        return int(total * 1.1)


if __name__ == '__main__':
    b = PriceCalculator(100, 10)
    b2 = PriceCalculator2(100, 10)

    assert b.price == b2.price

File: src/main/c/extract_method.cpp
#include <stdio.h>
#include <cassert>

// Before
class PriceCalculator
{
private:
    int unitPrice;
    int num;
public:
    PriceCalculator(int unitPrice, int num);
    int price();
};

PriceCalculator::PriceCalculator(int unitPrice, int num)
{
    this->unitPrice = unitPrice;
    this->num = num;
}

int PriceCalculator::price()
{
    return unitPrice * num * 1.1;
}

// After
class PriceCalculator2
{
private:
    int unitPrice;
    int num;
public:
    PriceCalculator2(int unitPrice, int num);
    int price();
    int addTax(int);
};

PriceCalculator2::PriceCalculator2(int unitPrice, int num)
{
    this->unitPrice = unitPrice;
    this->num = num;
}

int PriceCalculator2::price()
{
    return addTax(unitPrice * num);
}

int PriceCalculator2::addTax(int value)
{
    return value * 1.1;
}

int
main(void)
{
    PriceCalculator p(100, 10);
    PriceCalculator2 p2(100, 10);

    assert(p.price() == p2.price());

    return 0;
}
